### PR TITLE
Decouple Entur's organisation registry

### PR DESCRIPTION
**Pull request by testower**
* Move Entur implementation to sandbox
* Add new default implementation, reading from netex file

### CODE REVIEW

File: src/main/java/no/entur/uttu/organisation/NetexPublicationDeliveryFileOrganisationRegistry.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    return authorityRef;
+  }
+
+  private <T> T readFromSource(Source source) {
+    try {
+      JAXBElement<T> element = (JAXBElement<T>) getUnmarshaller().unmarshal(source);
+      return element.getValue();
+    } catch (JAXBException e) {
+      throw new RuntimeException(e);
```
Comment (testower) [inline src/main/java/no/entur/uttu/organisation/NetexPublicationDeliveryFileOrganisationRegistry.java:121]: Improve error handling

File: src/main/java/no/entur/uttu/organisation/OrganisationRegistry.java [hunk 2]
```diff
@@ -2,11 +2,36 @@
...
+   */
   Optional<GeneralOrganisation> getOrganisation(String id);
+
+  /**
+   * Check if the organisation represented by the operator reference id is a valid operator
+   * @param operatorRef The organisation id
+   * @return The id if it is a valid operator
+   * @throws CodedIllegalArgumentException if the organisation is not a valid operator
+   */
   String getVerifiedOperatorRef(String operatorRef);
```
Comment (testower) [inline src/main/java/no/entur/uttu/organisation/OrganisationRegistry.java:28]: This api is awkward? Maybe a good opportunity to fix

File: src/main/java/no/entur/uttu/organisation/OrganisationRegistry.java [hunk 3]
```diff
@@ -2,11 +2,36 @@
...
+
+  /**
+   * Check if the organisation represented by the operator reference id is a valid authority
+   * @param authorityRef The organisation id
+   * @return The id if it is a valid authority
+   * @throws CodedIllegalArgumentException if the organisation is not a valid authority
+   */
   String getVerifiedAuthorityRef(String authorityRef);
```
Comment (testower) [inline src/main/java/no/entur/uttu/organisation/OrganisationRegistry.java:36]: This api is awkward? Maybe a good opportunity to fix

File: src/main/java/no/entur/uttu/organisation/OrganisationRegistry.java [hunk 4]
```diff
@@ -2,11 +2,36 @@
 
 import java.util.List;
 import java.util.Optional;
+import no.entur.uttu.error.codedexception.CodedIllegalArgumentException;
 import org.rutebanken.netex.model.GeneralOrganisation;
 
+/**
+ * Represents an organisation registry used to populate authorities and operators references
+ */
 public interface OrganisationRegistry {
```
Comment (testower) [inline src/main/java/no/entur/uttu/organisation/OrganisationRegistry.java:11]: spi ?

**Review by vpaturet — APPROVED** [review]

**Pull request merged**